core-web/libs/sdk/experiments/src/lib/hooks/useExperiments.ts

```diff
@@ -0,0 +1,78 @@
+// Update the location of the DotExperiments instance when it changes
+import { useContext, useEffect } from 'react';
+
+import { isInsideEditor } from '@dotcms/client';
+
+import DotExperimentsContext from './DotExperimentsContext';
+
+import { EXPERIMENT_DEFAULT_VARIANT_NAME, EXPERIMENT_QUERY_PARAM_KEY } from '../shared/constants';
+
+/**
+ * `useExperiments` is a custom hook for updating the location of the DotExperiments instance when it changes.
+ *
+ * @returns {void}
+ */
+export const useExperiments = (): void => {
+    // retrieve experiment context
+    const experimentContext = useContext(DotExperimentsContext);
+    /**
+     * This effect adds a change listener to the experiment context, specifically updating the location.
+     * It doesn't have a meaningful return; it primarily triggers side-effects.
+     */
+    useEffect(() => {
+        if (!experimentContext || typeof document === 'undefined') {
+            return;
+        }
+
+        const insideEditor = isInsideEditor();
+        if (!insideEditor) {
+            const location = typeof window !== 'undefined' ? window.location : undefined;
+            if (experimentContext && location) {
+                experimentContext.locationChanged(location, experimentContext.customRedirectFn);
+            }
+        }
+    }, [experimentContext]);
```

Comment:
```suggestion
        const insideEditor = isInsideEditor();

        if (!insideEditor) {
            const location = typeof window !== 'undefined' ? window.location : undefined;

            if (experimentContext && location) {
                experimentContext.locationChanged(location, experimentContext.customRedirectFn);
            }
        }
    }, [experimentContext]);
```

Reply:
@fmontes I found this
```
"padding-line-between-statements": [
      "error",
      { "blankLine": "always", "prev": "const", "next": "*" },
      { "blankLine": "always", "prev": "*", "next": "if" },
      { "blankLine": "always", "prev": "if", "next": "*" },
      { "blankLine": "always", "prev": "*", "next": "const" }
    ]
```
add the spaces automatically.